src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -202,6 +206,15 @@ func (r *StatefulSetResource) rollingUpdate(
 		if err = r.queryRedpandaStatus(ctx, &adminURL); err != nil {
 			return fmt.Errorf("unable to query Redpanda ready status: %w", err)
 		}
+
+		adminURL.Path = "metrics"
+
+		if err = r.queryRedpandaUnderReplicatedPartition(ctx, &adminURL); err != nil {
```

Comment:
nit: this should be named compare/evaluate not query since it also contains comparison logic and returns error in case it it over the treshold

Reply:
Ok, renamed